lib/airbyte.rb

```diff
@@ -1,58 +1,79 @@
 # frozen_string_literal: true
 
 require_relative "airbyte/version"
-require "airbyte/base_client"
-
-require "airbyte/resources/workspace"
-require "airbyte/resources/source_definition"
-require "airbyte/resources/destination_definition"
-require "airbyte/resources/jobs"
-require "airbyte/resources/scheduler"
-require "airbyte/resources/source"
-require "airbyte/resources/destination"
-require "airbyte/resources/sync_connection"
+
 require "faraday"
 require 'ostruct'
 require 'typhoeus'
-require 'typhoeus/adapters/faraday'
 require 'connection_pool'
-
 module Airbyte
-  class Error < StandardError; end
-  # Your code goes here...
-  # def self.url
-  #   @url
-  # end
-  # def self.conn
-  #   @conn
-  # end
-  
+  class Error < StandardError; end  
+
   def self.configuration
     @configuration ||= OpenStruct.new
   end
   def self.options
     @options
   end
 
-  def self.connection
-    @connection
+  def self.connection_config_api
+    @connection_config_api
   end
+
+  def self.connection_airbyte_api
+    @connection_airbyte_api
+  end
+
   class Config < OpenStruct
   end
 
   def self.configure(&blk)
     @configuration = Airbyte::Config.new
 
     yield(configuration)
-    @connection = ConnectionPool::Wrapper.new(size: @configuration.pool || 32, timeout: @configuration.timeout || 10) do
-      connection = Faraday.new(:url => @configuration.host + ":#{@configuration.port|| 80}") do |builder|
+    @connection_config_api = ConnectionPool::Wrapper.new(size: @configuration.pool || 32, timeout: @configuration.timeout || 10) do
```

Comment:
is the timeout of 10 Sec recommended by Airbyte ?

Reply:
they have different timeouts for different operations. We explicitly pass 120 to @configuration.timeout while initializing gem.